docs/router.md

```diff
@@ -0,0 +1,204 @@
+# About Routers
+
+`gpii.express.router` is a subclass of [`gpii.express.middleware`](middleware.md) that wraps
+[the  Express `router` object](http://expressjs.com/en/4x/api.html#router), which is simply a container for a collection
```

Comment:
We need to talk about this again.  Routers *may* isolate groups of middleware by path (including `/` for "any path") or by method (including `use` for "any method"), but they don't have to do either.  Middleware may also be limited by method or path, so the comparison also falls apart their.

The only difference is that a router may contain other middleware.  Say for example that you have a piece of `counter` middleware, and a `responder` piece of middleware.  

```
components: {
  get: {
    type: "gpii.express.router",
    options: {
      method: "get",
      components: {
        counter: { type: "my.namespace.counterMiddleware", priority: "first" },
        worker: { type: "my.namespace.respondingMiddleware"}
      }
    }
  },
  post: {
    type: "gpii.express.router",
    options: {
      method: "post",
      components: {
        counter: { type: "my.namespace.counterMiddleware", priority: "first" },
        worker: { type: "my.namespace.posRespondingMiddleware"}
      }
    }
  },
  default: {
    type: "gpii.express.router",
    options: {
      components: {
        counter: { type: "my.namespace.counterMiddleware", priority: "first" },
        worker: { type: "my.namespace.defaultRespondingMiddleware"}
      }
    }
  }
}
```

Each instance of `my.namespace.counterMiddleware` is isolated from the other instances, and can safely count just the traffic heading to the next piece of middleware in the chain.

Anyway, I'm not saying this shouldn't be clearer, only that we still need to chat, obviously.